### PR TITLE
935 CI: add label trigger for extended testing

### LINKED ISSUE #935: Add special label to trigger extended tests
Instead of always waiting for integration with develop to run full CI for Intel or Nvida, add a label that will allow the user to trigger that extended test if they wish.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #935

### CODE REVIEW

**Comment by lifflander** [comment]
So this PR adds a trigger that will spawn extended testing if you add the label "runextendedtests"

**Comment by lifflander** [comment]
This should be sequenced after: https://github.com/DARMA-tasking/vt/pull/938. Do not merge until https://github.com/DARMA-tasking/vt/pull/938 gets merged so we generate this code.

File: .github/workflows/dockerimage-intel-18-ubuntu-mpich-extended.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: PR tests extended (intel 18.03, ubuntu, mpich)
```
Comment (pnstickne) [inline .github/workflows/dockerimage-intel-18-ubuntu-mpich-extended.yml:1]: Are these generated from a template (for managing changes), is not, can they be?

File: .github/workflows/dockerimage-intel-18-ubuntu-mpich-extended.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+on:
+  pull_request:
+    types:
+      - labeled
+
+jobs:
+
+  build:
+
+    if: github.event.label.name == 'runextendedtests'
```
Comment (pnstickne) [inline .github/workflows/dockerimage-intel-18-ubuntu-mpich-extended.yml:12]: First sentinel :D

**Pull request merged**